### ISSUE TITLE
How do I use this?

### ISSUE THREAD

**Issue by FatePGN** [LeoLombardi/tos-laimas-compass#1]
This seems different to how all the other addons work, how does one go about using this?

**Comment by LeoLombardi** [comment]
Hey Fate,

This is actually run as a stand-alone application that iterates with the
game through an add-on just like the others. I'm working on a installer,
and I'll let you know as soon as I have it released. =)

**Comment by dothuraka** [comment]
To be honest, Leo, I've tried to compile it, but gives me lots of errors when compiling, unfortunately :/.

**Comment by LeoLombardi** [comment]
Can you post your errors here, @dothuraka? That'll help me write a 'build
from scratch' guide. I've also finished a (extremely unoptimized) deploy
package, which should cover most needs until I finish a proper release.

**Comment by LeoLombardi** [comment]
(Also, thanks for let me know about it, I appreciate the feedback!)

**Comment by Sylex-FiKnight** [comment]
Does this work on windows 10 currently? With the current executable youve created I get

![launcherror](https://cloud.githubusercontent.com/assets/9436997/15298788/ccb44430-1b66-11e6-9f38-bc383e6f62ee.png)

**Comment by LeoLombardi** [comment]
Yes, my development environment is a Win10 machine. Thanks, I'll check into
it!

**Comment by Sylex-FiKnight** [comment]
No worries just wanted to let you know.  I moved it out of my download folder just to make sure it wasnt something odd with that and still generated the same error.

**Comment by dothuraka** [comment]
@LeoLombardi, tbh, the problem I see is the same shown from @Sylex-FiKnight.

Will check the executable, but this is possibly due to an error caused by different versions of needed instances. Maybe guiding which version of each package and every npm packages we do need, would be easier.

**Comment by dothuraka** [comment]
And the executable gives me the same error as @Sylex-FiKnight, the same error it gives me when npm start, also.

**Comment by LeoLombardi** [comment]
...Ok, *duh* moment. I forgot to set my .gitignore to exclude the
root /node_modules directory only - the one present inside the compiled
version should stay there. My apologies, I'm committing the fixed version
in a couple mins. I'll also include some disclaimers on the readme.md and a
guide to manually uninstall the whole thing.

**Comment by lbotinelly** [comment]
Done. @FatePGN , @dothuraka , @Sylex-FiKnight  - feel free to download this version, and let me know if it works for you.

**Comment by LeoLombardi** [comment]
Done. @FatePGN <https://github.com/FatePGN> , @dothuraka
<https://github.com/dothuraka> , @Sylex-FiKnight
<https://github.com/Sylex-FiKnight> - feel free to download this version,
and let me know if it works for you.

**Comment by Sylex-FiKnight** [comment]
That fixed it for me.  Now I will be testing it after I get home from work on getting my characters to show up.  Great work @LeoLombardi Thanks so much.

**Comment by Ttobas** [comment]
I tryed to run the .exe, it starts but really glitchy (frames are on each other, can't chose my username)
I also can't run the .bat file, got some errors ;w;

**Comment by LeoLombardi** [comment]
​Yep, I'm sorry for that. It *is* glitchy right now. Another user noticed
this behavior on the first run. After you kill something ingame or open the
map, close the app and open again. It should work (and stop fidgeting.)

The batch files are meant for development time, sorry - they shouldn't be
there. Thanks for pointing that out, I'll remove them.

**Comment by Sylex-FiKnight** [comment]
I was wrong new error. That didnt display when I first ran it but leads to the glitchy frames and such that @Ttobas experienced.
![new error](https://cloud.githubusercontent.com/assets/9436997/15302711/f49ca114-1b79-11e6-9cbb-5cefdf61bbd7.png)

**Comment by Ttobas** [comment]
Okay thanks a lot !

**Comment by LeoLombardi** [comment]
Also, I'm planning to run a major cleanup tonight and make the project more
release-worthy, The file containing your progress, tracker.json, can be
migrated between versions, so nothing will be lost.

**Comment by Ttobas** [comment]
@Sylex-FiKnight You got this problem because the program is already running, open your task manager (ctrl + shift + escape) and kill the "tos-laimas-compass.exe"

**Comment by Ttobas** [comment]
Oh and @LeoLombardi Do I need to start TOS before your compass or after ?

**Comment by LeoLombardi** [comment]
Sylex, try opening the task manager, and check if 'Electron' appears
several times, like this:

 [image: Inline image 1]

If that's the case, just kill the Electron processes. The program should
behave after that. And feel free to open an issue for that, so I don't
forget. =)

**Comment by lbotinelly** [comment]
Compass need to run first in order to create the add-on (or, if a previous
add-on set is present like Excrulon's, add itself to the list.) After that
you should be good to go.

**Comment by LeoLombardi** [comment]
@Sylex-FiKnight , try opening the task manager, and check if 'Electron' appears
several times, like this:

![2016-05-16 16_24_20-task manager](https://cloud.githubusercontent.com/assets/18711130/15303078/05b9e024-1b84-11e6-9d4a-91f27584e087.png)

If that's the case, just kill the Electron processes. The program should
behave after that. And feel free to open an issue for that, so I don't
forget. =)

**Comment by LeoLombardi** [comment]
@Ttobas I'd say before, since Compass need to create the add-on that'll serve as broker before the game starts. It'll also update the loader to support whatever plugins you already have in place.

**Comment by Ttobas** [comment]
@LeoLombardi I tryed to run it the way you're saying and it doesn't work. I can't see any new addon in the addon folder, nor in the data folder. Maybe it is because I actually installed my game in SteamLibrary in another hard drive ?

**Comment by LeoLombardi** [comment]
@Ttobas That may be the case - the app tries to detect the current steam repository. Let me simulate it here. I'll get back to you asap. =)

**Comment by Ttobas** [comment]
Sooo... by hand, I added the .lua file to my game, it is loaded, and inside the  laimascompass, it created the data.txt file, with the right content to it. BUT in the standalone application, I can't get anything.

A friend of mine just tryed the same and he has no problem such as steam in another hard drive or so. His addons correctly load automatically, but he has exactly the same not moving application..

**Comment by LeoLombardi** [comment]
@Ttobas would you mind asking your friend to shut down compass and opening
again after playing a little bit (opening map with M, or killing something?)

Also, in your case it won't do anything because the app itself can't locate
the file. I'm implementing a config.json that can be manually set, as well
as proper detection/folder selection by user.

Once again, thanks for the feedback!

**Comment by Ttobas** [comment]
I already asked him to do that, it doesn't work. 
(we are on windows 7 64bits)

**Comment by LeoLombardi** [comment]
@Ttobas I see - I'll set up a Win7 machine with similar configuration and give it a try.

**Comment by dothuraka** [comment]
@LeoLombardi Tbh is the folders previous to x86-x64 (windows, compiled) needed?

**Comment by Ttobas** [comment]
@dothuraka no, it seems they just are needed to compile the program.

**Comment by dothuraka** [comment]
That just makes my life somehow miserable. 

It would be rather good to have lesser 45mb from the software.

**Comment by Ttobas** [comment]
lol
Take in consideration it still a really early release of the program I think..

**Comment by LeoLombardi** [comment]
@dothuraka my bet is that @Ttobas is right - I'll eventually get to trimming the installation down, but first I want to make it work for all of you guys.

**Comment by LeoLombardi** [comment]
@dothuraka My apologies, this is a very early version - barely over a proof of concept. If it is stressing you out, don't bother with it; I can always let you know once it reaches a stable form.

**Comment by dothuraka** [comment]
Relax, Leo. We'll do it. I'm the one to apologize, because I'm not a great programmer, but I've tried to install in all the ways you can possibly think of.

**Comment by LeoLombardi** [comment]
@dothuraka I understand you only too well. I wanted to pick a pet project to learn NodeJS, express and Electron, so this implementation was all over the place until now - making it specially hard to get it to work. But we'll get there. ;)

**Comment by LeoLombardi** [comment]
@Ttobas I managed to hack together an interface to help you pick the game folder:

![2016-05-17 02_28_20-laima s compass](https://cloud.githubusercontent.com/assets/18711130/15313257/223c4396-1bd8-11e6-8bc6-91c0890797b0.png)

...Yes, that awful `open` button takes you to a folder selection dialog. Since you managed to make the addon work and it is generating the data.txt file, this *should* work. (I can't test because of maintenance.)

Let me know if this fixes your issue.

**Comment by dothuraka** [comment]
Well, it opens for me with console open and show some errors
![laimas](https://cloud.githubusercontent.com/assets/7873837/15322015/8d1a4e14-1c10-11e6-9208-fc97c09eae4f.JPG)

**Comment by LeoLombardi** [comment]
@dothuraka Oops, my bed. Exported an invalid .gitignore. Fixed now, please d/l and try again.

**Comment by Ttobas** [comment]
Hi ! 
i just tryed the new program and It opened the console (as you can see in @dothuraka's screenshot)
with as error: 
"bootstrap.min.js:6 Uncaught Error: Bootstrap's JavaScript requires jQuery"
"localhost/:1 GET http://localhost:9183/res/img/footer_back.png 404 (Not Found)"

**Comment by LeoLombardi** [comment]
Oh, those two exceptions are fine, as long as they're the only ones to pop.

**Comment by Ttobas** [comment]
Yep yep, It works !

**Comment by LeoLombardi** [comment]
@Ttobas Fixed! Your level should appear close to your character's name, like this:

![2016-05-17 12_30_14-laima s compass](https://cloud.githubusercontent.com/assets/18711130/15332216/bf51b54e-1c32-11e6-9332-1f1d3d71aeb7.png)

Once you do anything - kill a mob, explore a piece of the map - an event will appear under the Travel Log interface, like this:

![2016-05-17 12_30_28-laima s compass](https://cloud.githubusercontent.com/assets/18711130/15332253/e09d03a2-1c32-11e6-8a25-5933cd7b5537.png)

(The highlight item indicates which item caused the 'event' on that map.)

This is help the app detect your 'current location'. From there, click on the little white squigly line corresponding to another map... 

![2016-05-17 13_22_25-laima s compass](https://cloud.githubusercontent.com/assets/18711130/15332300/09943064-1c33-11e6-8219-ba04a864a154.png)

To have the route traced:

![2016-05-17 13_21_47-laima s compass](https://cloud.githubusercontent.com/assets/18711130/15332315/1d8f0a1c-1c33-11e6-9979-5c23234778fa.png)

**Comment by LeoLombardi** [comment]
@Ttobas 

> he has a data.txt file created, with content and it doesn't work. But I have not any data.txt created and on my computer it works...

Actually, you do - but as soon as it is created the app consumes it and removes the file. For a quick test, shut down the companion app and play the game - the file will be generated and stay there.

Now, about your friend - let's try a quick step-by-step:

- Run the companion app
- Play the game a bit
- Shutdown the companion app
- Open it again

If after that he still have an issue.. I need to investigate the possible cause.

**Comment by Ttobas** [comment]
Hi ! Thanks you for this new version !
We just tryed with my friend, and it ... still doesn't work. We've done exactly what you told me about and nope, no name...
Even if I can create the folder, lua file and SumAni.ipf, i'm almost sure that the error is due to his folder path. Because his data.txt file isn't "comsumed".
It's located inside his steam's folder, at the right place, with the right name. But his steam folder is located on another hard drive.
(we used the adding you put for my own problem)
And the steam folder itself is located in "E:\Program File (x86)\", the spacing here makes me feel like it's related... If you have any idea, we would love this a lot ! Thanks !

**Comment by Sylex-FiKnight** [comment]
Sorry for not updating on my end didnt have any time last night to get on and test anything. Hoping to have some time today.  Been sick and dealing with a sick baby aswell lol. . . been a fun few days

**Comment by LeoLombardi** [comment]
@Ttobas Now that error is interesting... seems like your friend doesn't have all the published files: Would it be possible to ask your friend to re-download the full repo and try again?

**Comment by LeoLombardi** [comment]
@Sylex-FiKnight Not a problem at all, real life should always come first. How's your baby doing?

**Comment by Sylex-FiKnight** [comment]
Hes much better now, but now mom is getting sick lol So it continues haha.

**Comment by Ttobas** [comment]
@LeoLombardi Don't worry, he re-download every new repo you upload to see if that's working.

**Comment by VeloxAzo** [comment]
Can't seem to get it working myself either.
https://vgy.me/t7MEGQ.png
https://vgy.me/61JPKP.png

I downloaded the master branch and I'm running the exe file in "tos-laimas-compass-win32-x64"

**Comment by Sylex-FiKnight** [comment]
Working on my desktop, now to copy my addon/data/patch folder to my surface2 pro and set up the options for the compass there.  Great work absolutely love it.   Something I noticed on the surface is it didnt automagically find the game folder in settings, so some may have to switch it there to their actual game folder.

**Comment by LeoLombardi** [comment]
> Hes much better now, but now mom is getting sick lol So it continues haha.

@Sylex-FiKnight Glad to hear that, hope you guys get well soon. =)

About automatic game folder resolution, I have only a 'quick and dirty' check in place atm. I'll refine it further.

@VeloxAzo Interesting - I'll check. Does the tracker works for you, though? Seems the error you were getting was preventing you from seeing the Advice module only.

I plan to work on Laima's Compass this weekend, so perhaps we'll have a more stable tool soon. =)

**Comment by Questionguy663** [comment]
Loving the addon so far, I was having problems getting it working when Tree of Savior was installed in programsx86 but upon moving it out to C:\Games everything loaded properly.  
The only thing I can't get working at the moment is the relay to mobile device. If I use mobile browser on the PC that I'm running the game from the URL works and everything looks good.  
If I do the exact same thing on my Windows Tablet the URL times out in Edge, Chrome, and PirateBrowser. :(

**Comment by Sylex-FiKnight** [comment]
They makes sense i forgot how locked down they make the program files
locations. Will adjust my steam install on my surface out of that directory
and we'll probably watch my surface work as flawless as my desktop

**Comment by Questionguy663** [comment]
Yep, definately move it to somewhere unprotected by Windows.

**Comment by Sylex-FiKnight** [comment]
Same questionguy from the tos reddit?

**Comment by Questionguy663** [comment]
That was me yes

**Comment by LeoLombardi** [comment]
@Questionguy663 no chewin' allowed on this repo, no sir! Be most welcome. =)

I'll hopefully finish setting up a machine with standard configurations this weekend so I can test stuff before releasing a new version. 

Meanwhile, new version is cooking, screenshot:

![2016-05-20 23_10_03-laima s compass](https://cloud.githubusercontent.com/assets/18711130/15446056/0ec86f96-1ee0-11e6-9ea5-bbe4f292b290.png)

**Comment by Questionguy663** [comment]
That looks really nice!  
I came across another couple of things, I had my level showing up at one point and then it disapeared could have been a level up, it could have been when I restarted my client or changed maps so that could be something to be on the lookout for!  

Also I was wondering if we could get a "run on Windows Startup" option, I noticed at the mod is picking up information as we are walking around and killing rather than retroactively looking at the adventure journal it would be a shame if I played for ages without remembering to turn it on with my computer.  

I know I could setup some sort of windows task manager thing to do this but I was just wondering if that could be a feature. :D

**Comment by Sylex-FiKnight** [comment]
No i wasn't attacking him wanted to thank him because of his post i was
able to find this project.
But back to helping those it's not working for. If it's in program files or
program files(x86) it won't work correctly without setting security on it
but you risk other items installed our self installing a lot of leeway.
Just move steam out to the C:\Steam and it will fix the problems Most are
seeing.

**Comment by Questionguy663** [comment]
Ah so I was right then :D

**Comment by Sylex-FiKnight** [comment]
Yup you definitely were.  Great find. Thanks for this awesome utility Leo.

**Comment by Questionguy663** [comment]
Once I get it to run on startup (either through the program or a windows task) and once I can stream it over to my mobile device i'll be in heaven. I used to run three monitors which was great for having extra browsers open ect but now that i'm on one monitor this will be extra helpful.

**Comment by Sylex-FiKnight** [comment]
Getting it to run on start is fairly easy. The streaming to a mobile device
lol not as such lol.

**Comment by Questionguy663** [comment]
Well If the URL that is already provided inside the program for that _is_ already working (works if I do it on the same computer) then it's probably something wrong on my end, but I already allowed it through my firewall so I'm not sure what else to try next.

**Comment by Sylex-FiKnight** [comment]
Let me take a look at something i may be able to help with that.  It's your
phone on your wifi or your data plan when you try

**Comment by Questionguy663** [comment]
I don't actually own a smartphone, I've got a Windows Tablet [This one](http://www.argos.co.uk/static/Product/partNumber/4967297.htm)

**Comment by Sylex-FiKnight** [comment]
@LeoLombardi what file is that URL link supposed to reference?

**Comment by Questionguy663** [comment]
It's not important, To answer your question I am not on a data plan i'm on my wifi as it is a tablet not a phone. I should have said that the first time around sorry!

**Comment by Questionguy663** [comment]
woops I wrote a reply because I thought you asked me what my link was, haha oops!

**Comment by Sylex-FiKnight** [comment]
Lol figured that out. But with the html file that url references. I think i
can help get it on your tablet.

**Comment by LeoLombardi** [comment]
> No i wasn't attacking him wanted to thank him because of his post i was
able to find this project.

@Sylex-FiKnight I know, I was just joking. =)

> Also It was nice to see someone who my old thread helped

The dynamic addon loader I'm using on this project is basically the same from another project you posted as well - and I got in touch with the dev and he gave me his blessings to use it. So you see, you contributed to this project as well. ;)

About the streaming - this is actually something easy to pinpoint the issue. The mobile link will work as long as both your gaming machine and your mobile device are on the same network. For example, in my case I have a wireless modem where I also have my gaming machine hardwired to it. Both belong to the same 'home' network.

The link shown in the application is nothing more than your physical machine name. The application actually implements a tiny webserver, so when you type that URL on your mobile browser you're accessing a web app on your gaming machine.

**Comment by Questionguy663** [comment]
Ok! let's get this troubleshooting going! :D 
Once i'm file exploring (Windows 10) and I click nextwork, I'm told "Network Discovery and file sharing are turned off network computers and devices are not visible click to change"  
First thing's first, will network discovery need to be on?  

I'm glad to hear more things about how what I was doing helped, I did get one appology from one of the people who wanted my thread shut down, I might find out if I can get something done about getting a new one, I have even been thinking about a new helpful feature.

**Comment by Sylex-FiKnight** [comment]
If both are win10 right click the start button or long press and hold via
touch and select system. Then under computer name, domain and workgroup
settings. Make sure workgroup matches on both. If it doesn't click change
settings and change the workgroup name. Out will require a reboot after tho.

**Comment by LeoLombardi** [comment]
> will network discovery need to be on?

No, not at all. As long as your firewall is configured to allow incoming access to TCP port 9183, you're good to go; that's what the Windows pop-up that comes up when you first try to run the app is supposed to allow.

About the thread, don't take it to heart. Developers can sometimes be very protective of their own creations, even opensource ones.

**Comment by Questionguy663** [comment]
@Sylex-FiKnight I'll give both of the systems a check right now

@LeoLombardi I might actually need to clear that TCP port through my router as well as it can be fussy, it couldn't hurt to try!

The weird thing that caused half the problem was the addon that I shared with friend of authors permission sure, but the author kept on ignoring the way I had the URL from her friend and that it was up there as a committed release. There was no way I could have had the URL for that github without the friend, i'm not made of magic links that find their own way to me! and the friend said it was ok because it was a release version that they had permission for me to add it to my topic!

**Comment by LeoLombardi** [comment]
> There was no way I could have had the URL for that github without the friend

Well, my personal opinion - If you don't want people to find, don't put it on a public repo. =)

**Comment by Questionguy663** [comment]
Yeah, one thing that was mentioned that that secret repo's cost money, but if it were me i'd make two free ones and not share one out ever :P

**Comment by LeoLombardi** [comment]
Well, GitHub is like that, but if you want a private repo you can just install the service on your own home machine, and control your source locally (and, hopefully, also back it up...)

**Comment by Questionguy663** [comment]
and cloud save it to the One Drive and Google Drive! Added the ports to my Port Foward, giving the router a restart for safe measure, and then taking it for a quick test!

**Comment by LeoLombardi** [comment]
![caeca14caf425c6de80bd94f29f63f0a1c5197fecabd50b1b1d916a79d9b8685](https://cloud.githubusercontent.com/assets/18711130/15446478/f38fd5a6-1eef-11e6-93a1-8c6f8eb48008.jpg)

**Comment by Questionguy663** [comment]
I wonder if having a computer name with a hyphen in it is causing problems!  
DESKTOP-GPOFEFM:9183  
Thats my link.

**Comment by LeoLombardi** [comment]
@Questionguy663 hyphenated names shouldn't be an issue either. If I had to bet, I'd say your router may be segmented your network as a security measure, preventing two machines from talking to each other locally.

**Comment by Questionguy663** [comment]
That's a shame, I have no idea how i'd go about fixing that.

**Comment by Questionguy663** [comment]
Lol normally I wouldn't double post, but accidental tech wizardry! Turned on network discovery and **BAM**. Everything worked as intended. Meaning I didnt not have to run the mobile browser on the **desktop** from the Tablet, I could use the Tablet's Mobile Browser.  
Apparently you can't use a local computers URL unless you can see that computer on your network (on my network atleast :P)

**Comment by LeoLombardi** [comment]
@Questionguy663 Oh my! At least it is working now for you. I'll keep that hint handy, could swear that network discovery wouldn't affect it at all. Live and learn. =)

About contributions - good idea, maybe an extra tab or an easter egg. I'l think of something. At least worth a mention on readme.md. ;)

**Comment by Questionguy663** [comment]
Oh i just meant on the website, nowhere grant :P  

Here is my other thing I was just dying to tell you (I'm replying just two minutes after you!  

DESKTOP-GPOFEFM:9183 current format, works only on mobile browsers, on regular browser it does a search with your default search engine.  

http://desktop-gpofefm:9183/#/log this format works in **all** browsers without having to do anything. Meaning Tablets, Phones, **Laptops** second **Desktop is the now possible!

**Comment by LeoLombardi** [comment]
@Questionguy663 Oh, that makes sense, you have to specify the protocol (http) in order to avoid artificial stupidity from the browser. =)

I'll fix the way the info is being shown. Thanks!

BTW, the pathing tool is working directly from github! Care to test?

https://rawgit.com/LeoLombardi/tos-laimas-compass/master/tos-laimas-compass-win32-x64/resources/app/index.html#/route/1001/3620

**Comment by Questionguy663** [comment]
Tried a couple of simple routes that I know like 'Klaipeda' to 'Miner Village' and it reminded me of one thing I noticed when using the program.  
There's a lot of places (and monsters) that you have to think about a bit when you look at them because they are the RAW engrishy half translated name.  
I.E on Klaidepa to Miner Village there is **f_siauliai_2** which is East Siauliai Woods (which i feel would be impossible to guess if it was for a large distance pathing  
I.E. for monsters at Igti Coast there is **Rebusbunny_bow_red** which a Red Lepusbunny Assassin, not the worst example though :P  

Other than the names though it looks good, is there anything specific you would like me to test, if so i'll look some more :D

**Comment by LeoLombardi** [comment]
I know, I need to refine the import script. I'm getting there, but sources are kind of all around. =)

**Comment by Questionguy663** [comment]
Oh sorry I didn't mean anything by it :(  
I like all the extra info, feild map or dungeon, if it has a teleport. It's very cool!

**Comment by LeoLombardi** [comment]
Oh, but the OCD dev in me looks at those names and cringe... ;)

**Comment by Questionguy663** [comment]
Would you like me to write down some of the names that aren't done yet and what they should be? I can see what I can find if that would be helpful

**Comment by LeoLombardi** [comment]
@Questionguy663 I wouldn't worry too much about that - I'll dig through the source files one more this weekend, and try to fill the missing bits. But thanks for the kind offer. =)

**Comment by JourneyOver** [comment]
is this still being worked on??
